gettsim/taxes/zve.py

```diff
@@ -318,25 +319,64 @@ def vorsorge2010(tax_unit, tb):
     # calculate x% of relevant employer and employee contributions
     # then subtract employer contributions
     # also subtract health + care + unemployment insurance contributions
-    altersvors2010 = ~tax_unit["child"] * (
-        (0.6 + 0.02 * (np.minimum(tb["yr"], 2025) - 2005)) * (12 * rvbeit_vors)
-        - (12 * 0.5 * rvbeit_vors)
-    )
+    altersvors2010 = ~tax_unit["child"] * vorsorge_year_faktor(tb["yr"]) * (
+        12 * rvbeit_vors
+    ) - (12 * 0.5 * rvbeit_vors)
+
     # These you get anyway ('Basisvorsorge').
     sonstigevors2010 = 12 * (tax_unit["pvbeit"] + 0.96 * tax_unit["gkvbeit"])
     # maybe add avbeit, but do not exceed 1900€.
     sonstigevors2010 = np.maximum(
         sonstigevors2010,
         np.minimum(sonstigevors2010 + 12 * tax_unit["avbeit"], tb["vors_sonst_max"]),
     )
+    return altersvors2010.astype(int) + sonstigevors2010.astype(int)
 
-    vorsorge2010 = altersvors2010 + sonstigevors2010
 
-    return vorsorge2010.astype(int)
+def vorsorge2005(tax_unit, tb):
+    """ Vorsorgeaufwendungen pre 2010
+    Pension contributions are accounted for up to €20k.
+    From this, a certain share can actually be deducted,
+    starting with 60% in 2005.
+    Other deductions are just added, up to a ceiling of 2400 p.a. for standard employees.
 
+    Background: https://bit.ly/32oqCQq
+    """
+    tb["max_altersvors_2005"] = 20000
```

Comment:
This parameter is not fixed from 2015 on: Artikel 5 G. v. 22.12.2014 BGBl. I S. 2417, see https://www.buzer.de/gesetz/4499/al46488-0.htm

Reply:
You are right, but isn't this essentially irrelevant after 2010 since the new ruling is always better than the old one?

Reply:
I don't know if it is always better. Still, it might become again relevant in a reform or counterfactual. Probably depends on Riester stuff too.